app/controllers/discourse_assign/assign_controller.rb

```diff
@@ -44,11 +44,27 @@ def unassign
     end
 
     def assign
-      reassign_or_assign_target(action: "assign")
-    end
+      target_id = params.require(:target_id)
+      target_type = params.require(:target_type)
+      username = params.permit(:username)['username']
+      group_name = params.permit(:group_name)['group_name']
 
-    def reassign
-      reassign_or_assign_target(action: "reassign")
+      assign_to = username.present? ? User.find_by(username_lower: username.downcase) : Group.where("LOWER(name) = ?", group_name.downcase).first
+
+      raise Discourse::NotFound unless assign_to
+      raise Discourse::NotFound if !Assignment.valid_type?(target_type)
+      target = target_type.constantize.where(id: target_id).first
+      raise Discourse::NotFound unless target
+
+      # perhaps?
+      #Scheduler::Defer.later "assign topic" do
+      assign = Assigner.new(target, current_user).send("assign", assign_to)
```

Comment:
Why are you using `send` when you can call the `assign` method directly?

```suggestion
      assign = Assigner.new(target, current_user).assign(assign_to)
```

Reply:
This was left over from an old method. Good catch, I will remove.